tests/test_autoregressive_generator.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+#
+
+
+import pytest
+import torch
+
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from tests.tests_constants import local_model_path
+
+from self_speculation.generator_base import GenerationConfig
+from self_speculation.autoregressive_generator import AutoRegressiveGenerationStrategy
+
+@pytest.fixture
+def model_and_config():
+    """Fixture to create a model and a generation configuration."""
+    # Initialize a lightweight version of LlamaForCausalLM if possible
+    tokenizer = AutoTokenizer.from_pretrained(local_model_path)
+    model = AutoModelForCausalLM.from_pretrained(
+        local_model_path,
+        use_safetensors=True,
+        device_map="mps",
```

Comment:
Can we replace the `mps` with `auto`? 
I was running on a Linux server with a GPU, and `auto` automatically detects the GPU. I think when running on MacBook, it should automatically detect `mps`, but you can give it a try.

```suggestion
        device_map="auto",
```